docs/source/plugin/test.rst

```diff
@@ -0,0 +1,323 @@
+it registers everything into its ``message_sent`` list::
+
+    from sopel.tests import rawlist
+
+
+    def test_my_command(bot):
+        bot.say('Hi!', '#channel')
+        assert len(bot.backend.message_sent) == 1
+        assert bot.backend.message_sent == rawlist(
+            'PRIVMSG #channel :Hi!',
+        )
+
+.. seealso::
+
+    For more information about the :func:`~sopel.tests.rawlist` function,
+    see the `Checking the bot's output`_ section.
+
+Test Server and test Users
+--------------------------
+
+Alright, you have a properly setup test bot, now it's time for the last parts
+of your test setup: a test server with test users. As usual, there are fixtures
+to help you. The :func:`~sopel.tests.pytest_plugin.ircfactory` can be used to
+create a test server, and the :func:`~sopel.tests.pytest_plugin.userfactory`
+can create test users::
+
+    @pytest.fixture
+    def irc(bot, ircfactory):
+        return ircfactory(bot)
+
+    def test_my_command(bot, irc, userfactory):
+        user = userfactory('MyNick')
+        irc.pm(user, '.mycommand arg')
+
+        assert bot.backend.message_sent == rawlist(
+            'PRIVMSG MyNick :Command answer to a regular user.'
+        )
+
+    def test_my_command_owner(bot, irc, userfactory):
+        owner = userfactory('OwnerNick')
+        irc.pm(owner, '.mycommand arg')
+
+        assert bot.backend.message_sent == rawlist(
+            'PRIVMSG MyNick :Command answer to my owner.'
+        )
+
+As usual, you can create custom fixtures for the test server and for your test
+users, for example, one for a regular user, and one for the owner::
+
+    @pytest.fixture
+    def user(userfactory):
+        return userfactory('MyNick')
+
+
+    @pytest.fixture
+    def owner(userfactory):
+        return userfactory('OwnerNick')
+
+Channel messages
+................
+
+A bot can join channels, and so does your test bot: by using the test server,
+you can make the bot join a channel, or add new users to a channel while the
+bot is already in::
+
+    def test_my_command(bot, irc, user, owner):
+        # bot joins #channel with the owner in it
+        irc.channel_join('#channel', users=[owner])
+
+        # user joins #channel after
+        irc.join(user, '#channel')
+
+        # clear messages on join
+        bot.backend.clear_message_sent()
+
+        # user talks into a channel
+        irc.say(user, '.mycommand arg')
+
+        assert bot.backend.message_sent == rawlist(
+            'PRIVMSG #channel :MyNick: my reply into a channel.'
+        )
+
+You can automate this setup within your fixture::
+
+    @pytest.fixture
+    def irc(bot, user, owner, ircfactory):
+        test_server = ircfactory(bot)
+        # auto-join channels
+        test_server.channel_join('#channel', users=[owner, user])
+        # clear messages on join
+        bot.backend.clear_message_sent()
+        return test_server
+
+And now you are all setup to test your plugin's commands and rules!
```

Comment:
```suggestion
And now you are all set up to test your plugin's commands and rules!
```

This compound word is never used correctly even by native speakers, I swear. 👀

Reply:
And I swear I have checked once or twice for another occurrence, but not that one. 😁